### PR TITLE
F multicolumn combobox (~2940)

### PR DESCRIPTION
**Pull request by MaxPatyk**

### CODE REVIEW

File: release-notes.xml [hunk 1]
```diff
@@ -69,4 +69,7 @@
   <ChangeSet guid="4f405467-cea8-461a-b87e-bdc622cbb6fe">
     <Change type="feature">Adds an api to return the document inside a workflow to the selected return folder.</Change>
   </ChangeSet>
+  <ChangeSet>
+	<Change type="feature">Adds a new window to forwarded a document with the multicolumn-combobox.</Change>
```
Comment (simplicbe) [inline release-notes.xml:73]:
```suggestion
    <Change type="feature">Adds a new window to forwarded a document with the multicolumn-combobox.</Change>
```

File: src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            RaisePropertyChanged(nameof(MultiItemboxItems));
+            this.dictParams = dictParams;
+            RaisePropertyChanged(nameof(DictParams));
+            RaisePropertyChanged(nameof(CommentText));
+        }
+
+        public ObservableCollection<IMultiSelectionComboBoxItem> MultiItemboxItems { get; set; } = new ObservableCollection<IMultiSelectionComboBoxItem>();
+        public Dictionary<string, string> DictParams { get => dictParams; set => dictParams = value; }
+        public string CommentText { get; set; }
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs:22]: comments

File: src/Simplic.FileStructure.Workflow.UI/MultiItemBox.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using System.Windows;
+
+namespace Simplic.FileStructure.Workflow.UI
+{
+    /// <summary>
+    /// Interaction logic for MultiItemBox.xaml
+    /// </summary>
+    public partial class MultiItemBox : DefaultRibbonWindow
+    {
+        public MultiItemBox(Dictionary<string, string> dictParams)
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/MultiItemBox.xaml.cs:13]: comment

File: src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs (hunk 2 above)
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs:22]: spacing

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 4]
```diff
@@ -91,90 +96,100 @@ public static GridInvokeMethodResult ForwardTo(GridFunctionParameter parameter)
                 return GridInvokeMethodResult.NoGridRefresh();
             }
 
-            var itemBox = (AsyncGridItemBox)ItemBoxManager.GetItemBoxFromDB("IB_Document_Workflow_User");
-            itemBox.SetPlaceholder("WorkflowId", parameter.GetSelectedRowsAsDataRow().FirstOrDefault()["WorkflowId"].ToString());
-            itemBox.ShowDialog();
-            int targetUserId = 0;
-            Guid? workflowOrganzisationId = null;
+            Dictionary<string, string> dictParams = new Dictionary<string, string>();
+            dictParams.Add("[WorkflowId]", "0a2e6fa8-2ee9-4d2e-a6e1-48cb2bf66078");
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:100]: dont use magical strings

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 5]
```diff
@@ -91,90 +96,100 @@ public static GridInvokeMethodResult ForwardTo(GridFunctionParameter parameter)
...
-            if (itemBox.SelectedItem == null)
-                return new GridInvokeMethodResult { RefreshGrid = false };
+            var win = new MultiItemBox(dictParams);
+            win.ShowDialog();
 
-            if ((string)itemBox.GetSelectedItemCell("InternalType") == "User")
-                targetUserId = (int)itemBox.GetSelectedItemCell("Ident");
-            else
-                workflowOrganzisationId = (Guid)itemBox.GetSelectedItemCell("Guid");
+            var test2 = win.DataContext;
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:105]: test should not the name of a variable

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 6]
```diff
@@ -91,90 +96,100 @@ public static GridInvokeMethodResult ForwardTo(GridFunctionParameter parameter)
...
+                var comment = new Framework.Extension.InstanceDataCommentModel
                 {
-                    workflowOperationService.ForwardTo(workflowOperation);
-                }
-                catch (DocumentWorkflowException ex)
+                    Comment = commentText,
+                    UserGroupVisibility = Visibility.Hidden,
+                    UserId = sessionService.CurrentSession.UserId,
+                    InstanceDataGuid = Guid.NewGuid(),
+                    StackGuid = Guid.Parse("12c9b95b-bd33-4fa0-9ca1-05e11122018c") // TODO: Remove magic string
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:140]: remember the todo

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 7]
```diff
@@ -184,85 +199,102 @@ public static GridInvokeMethodResult ForwardCopyTo(GridFunctionParameter paramet
...
-            itemBox.ShowDialog();
+            Dictionary<string, string> dictParams = new Dictionary<string, string>();
+            dictParams.Add("[WorkflowId]", "0a2e6fa8-2ee9-4d2e-a6e1-48cb2bf66078");
 
-            if (itemBox.SelectedItem == null)
-                return new GridInvokeMethodResult { RefreshGrid = false };
+            var win = new MultiItemBox(dictParams);
+            win.ShowDialog();
 
+            var test2 = win.DataContext;
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:213]: test is not a valid variable name

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 8]
```diff
@@ -184,85 +199,102 @@ public static GridInvokeMethodResult ForwardCopyTo(GridFunctionParameter paramet
...
+                        DocumentId = documentId,
+                        DocumentPath = documentPathId,
+                        UserId = sessionService.CurrentSession.UserId,
+                        TargetUserId = targetUserId,
+                        CreateDateTime = DateTime.Now,
+                        UpdateDateTime = DateTime.Now,
+                        ActionName = "forward",
+                        WorkflowId = workflowId,
+                        Guid = Guid.NewGuid(),
+                    };
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:269]: a lot of this seems to be implemented twice, is it worth to move it to a seperate method?

**Review by MEichhoff — CHANGES_REQUESTED** [review]

**Comment by MaxPatyk** [comment]
image

File: src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+using Simplic.UI.MVC;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Simplic.FileStructure.Workflow.UI
+{
+    /// <summary>
+    /// Represents a viewmodel for managing multi-itembox-items.
+    /// </summary>
+    public class MultiItemBoxViewModel : ViewModelBase
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/MultiItemBoxViewModel.cs:11]: same with renaming

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 10]
```diff
@@ -158,122 +104,132 @@ public static GridInvokeMethodResult ForwardTo(GridFunctionParameter parameter)
...
+                    workflowOperationService.ForwardCopyTo(workflowOperation);
+                }
+                catch (DocumentWorkflowException ex)
+                {
+                    Log.LogManagerInstance.Instance.Error("Could not forward document in workflow", ex);
 
-            Guid? workflowOrganizationId = null;
-            int targetUserId = 0;
+                    // TODO: Add localization
+                    MessageBox.Show("Workflow für Zielbenutzer nicht gefunden.", "Workflow nicht gefunden", MessageBoxButton.OK, MessageBoxImage.Information);
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:131]: remember the todo

File: src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs [hunk 11]
```diff
@@ -158,122 +104,132 @@ public static GridInvokeMethodResult ForwardTo(GridFunctionParameter parameter)
...
-                    workflowOperationService.ForwardCopyTo(workflowOperation);
-                }
-                catch (DocumentWorkflowException ex)
+                foreach (var row in parameter.GetSelectedRowsAsDataRow())
                 {
-                    Log.LogManagerInstance.Instance.Error("Could not forward document in workflow", ex);
+                    var documentId = (Guid)row["Guid"];
+                    var documentPathId = (Guid)row["DocumentPathId"];
+                    // The grid needs a the column workflow id 
+                    var workflowId = (Guid)fileStructureDocumentPathService.Get(documentPathId).WorkflowId;
```
Comment (MEichhoff) [inline src/Simplic.FileStructure.Workflow.UI/WorkflowApplicationHelper.cs:198]: possible nre?

**Review by MEichhoff — CHANGES_REQUESTED** [review]

**Review by MEichhoff — APPROVED** [review]

**Pull request merged**